api/v2/api.go

```diff
@@ -124,8 +124,6 @@ func NewAPI(
 	openAPI.SilenceGetSilencesHandler = silence_ops.GetSilencesHandlerFunc(api.getSilencesHandler)
 	openAPI.SilencePostSilencesHandler = silence_ops.PostSilencesHandlerFunc(api.postSilencesHandler)
 
-	openAPI.Logger = func(s string, i ...interface{}) { level.Error(api.logger).Log(i...) }
```

Comment:
After reading through the openapi code, it seems that it isn't used anywhere because we're not using `*github.com/prometheus/alertmanager/api/v2/restapi.Server` directly.
In any case the code here didn't work as expected, it should have been `level.Error(api.logger).Log("msg", fmt.Sprintf(s, i...))`.